coordinator/src/collaborative_revert.rs

```diff
@@ -28,254 +29,242 @@ use dlc_manager::Storage;
+            .context("Could not calculate settlement amount")?;
+
+        let dlc_channel_reserved_tx_fees = estimate_subchannel_reserved_tx_fees(
+            fund_txo_sat,
+            ln_inbound_liquidity_sat,
+            ln_outbound_liquidity_sat,
+            unspendable_punishment_reserve_sat,
+            position.trader_margin as u64,
+            position.coordinator_margin as u64,
+        )?;
+        let dlc_channel_reserved_tx_fees_per_party = dlc_channel_reserved_tx_fees as f64 / 2.0;
+
+        fund_txo_sat as i64
+            - ln_inbound_liquidity_sat as i64
+            - unspendable_punishment_reserve_sat as i64
+            - counterparty_settlement_amount as i64
+            - dlc_channel_reserved_tx_fees_per_party as i64
+    };
 
     // Coordinator's amount is the total channel's value (fund_value_satoshis) whatever the taker
     // had (inbound_capacity), the taker's PnL (settlement_amount) and the transaction fee
-    let coordinator_amount = match trade_settled {
-        false => {
-            sub_channel.fund_value_satoshis as i64
-                - (channel_details.inbound_capacity_msat / 1000) as i64
-                - settlement_amount as i64
-                - (dlc_channel_fee as f64 / 2.0) as i64
-                - channel_details.unspendable_punishment_reserve.unwrap_or(0) as i64
-        }
-        true => {
-            sub_channel.fund_value_satoshis as i64
-                - (channel_details.inbound_capacity_msat / 1000) as i64
-                - channel_details.unspendable_punishment_reserve.unwrap_or(0) as i64
-        }
-    };
-    let trader_amount = sub_channel.fund_value_satoshis - coordinator_amount as u64;
 
-    let fee = weight_to_fee(
-        COLLABORATIVE_REVERT_TX_WEIGHT,
-        revert_params.fee_rate_sats_vb,
-    )
-    .expect("To be able to calculate constant fee rate");
-
-    tracing::debug!(
-        coordinator_amount,
-        fund_value_satoshis = sub_channel.fund_value_satoshis,
-        inbound_capacity_msat = channel_details.inbound_capacity_msat,
-        settlement_amount,
-        dlc_channel_fee,
-        inbound_capacity_msat = channel_details.inbound_capacity_msat,
-        outbound_capacity_msat = channel_details.outbound_capacity_msat,
-        trader_margin = position.trader_margin,
-        coordinator_margin = position.coordinator_margin,
-        position_id = position.id,
-        "Collaborative revert temporary values"
-    );
+    let trader_amount = subchannel.fund_value_satoshis - coordinator_amount as u64;
+
+    let fee = weight_to_fee(COLLABORATIVE_REVERT_TX_WEIGHT, fee_rate_sats_vb)
+        .expect("To be able to calculate constant fee rate");
 
     let coordinator_address = node.get_unused_address();
     let coordinator_amount = Amount::from_sat(coordinator_amount as u64 - fee / 2);
     let trader_amount = Amount::from_sat(trader_amount - fee / 2);
 
-    // TODO: check if trader still has more than dust
```

Comment:
❓ did you check that? 😅

Reply:
I meant to move this TODO because I don't think we should check that here. The coordinator should forward their opinion on the values to the app and the app should check those values and build the transaction accordingly. If they find that either of the amounts is below dust, the _app_ should build a transaction with a single output.

Reply:
I've tackled the TODO the way I outlined above.